_posts/2015-12-15-Automating-Testflight-Deploys.markdown

```diff
@@ -0,0 +1,48 @@
+
+We use a Makefile, I know that Fastlane provides an awesome tool in the form of [Fastlane Lanes](https://github.com/fastlane/fastlane#features) - but we're pretty happy with a Makefile, they're the simplest tool that does what we need.
+
+I wanted to lower the barrier for us shipping betas, so I opted to add another build step in the CI process. This step checks what branch is it, and if it's the beta branch, grab the certs, then deploy.
+
+``` sh
+deploy_if_beta_branch:
+	if [ "$(LOCAL_BRANCH)" == "beta" ]; then make certs; make ipa; make distribute; fi
+```
+
+`make certs` is really simple, it runs: `bundle exec match appstore --readonly`  which and pulls metadata from a [MatchFile](https://github.com/artsy/eigen/blob/9eaf9082ebdcdf75f12ad2804260587e01526f2d/fastlane/Matchfile). This means we can sign app store builds on CI.
+
+If you don't know what the `bundle exec` prefix is, I'd recommend reading my guide on the CocoaPods website for [Gemfile](https://guides.cocoapods.org/using/a-gemfile.html)s.
+
+The next step is generating an ipa, we do this with [gym](https://github.com/fastlane/gym) via `make ipa` which looks like this:
+
+``` sh
+ipa: set_git_properties change_version_to_date
+	bundle exec gym
+```
+
+It executes some make tasks to ensure we know what git commit each build is, and we use the date to provide a faux-[semver](http://semver.org) for apps.
+
+Gym will build our app, according to our [GymFile](https://github.com/artsy/eigen/blob/9eaf9082ebdcdf75f12ad2804260587e01526f2d/fastlane/GymFile). Nothing too surprising in there. It will output an [ipa](http://apple.stackexchange.com/questions/26550/what-does-ipa-stand-for) and a [dsym](http://stackoverflow.com/questions/3656391/whats-the-dsym-and-how-to-use-it-ios-sdk) that `make distribute` can handle.
+
+`make distribute` is a pretty easy one, we generate a CHANGELOG via Ruby, then run the command `bundle exec pilot upload -i build/Artsy.ipa`, it will ship it to iTunes Connect after configuration from the [AppFile](https://github.com/artsy/eigen/blob/9eaf9082ebdcdf75f12ad2804260587e01526f2d/fastlane/AppFile). This is great, but it goes one better. It will, by default, run a synchronous check for whether the App has finished processing.
+
+{% expanded_img /images/2015-12-15-Automating-Testflight-Deploys/ci-itunes-screenshot.png %}
+
+This is awesome. I'd like to add a Slack message to tell us that it's shipped too. We've not entirely moved all of our apps to TestFlight, this is our first experiment in the space, we've been really happy with Hockey, and still are. However, without trying new things we'll never be able to know what we should consider internal best practices.
```

Comment:
Good conclusion – worth nothing that Slack is totally something that Fastlane can do (and we do for Eidolon).